epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyModelWithView.java

```diff
@@ -0,0 +1,58 @@
+package com.airbnb.epoxy;
+
+import android.support.annotation.LayoutRes;
+import android.view.View;
+import android.view.ViewGroup;
+
+/**
+ * A model that allows its view to be built programmatically instead of by inflating a layout
+ * resource. Just implement {@link #buildView} so the adapter can create a new view for this model
+ * when needed.
+ * <p>
+ * {@link #getViewType()} is used by the adapter to know how to reuse views for this model. If it is
+ * left unimplemented then the generated model will include an implementation which returns a value
+ * based on the generated model's name. This means that all models of that type should be able to
+ * share the same view, but the view won't be shared with models of any other type.
+ * <p>
+ * The generated view type will be negative so that it cannot collide with values from layout
+ * resources, which are used in normal Epoxy models. However, it is possible for generated view
+ * types to collide if models from different libraries are used together, since the model processor
+ * can only guarantee that view types are unique within a library. If you need to use {@link
+ * EpoxyModelWithView} models from different libraries or modules together you can define a view
+ * type manually to avoid this small collision chance. A good way to manually create value is by
+ * creating an R.id. value in an ids resource while, which will guarantee a unique value.
```

Comment:
yep, something like that would definitely work. The main issue is this can't work across multiple modules. They would instantly clash since they all start at -1 and go down. By using a hash we at least have a pretty high chance of not having a collision. The other thing I considered is that a hash of the class name is reproducible, whereas something like this could arbitrarily result in a different value if more model classes are added/deleted/implementation changes or whatever. it seems useful to have a fairly reproducible viewtype so if people do have a collision it is reproducible.

On the other hand, if we did it this way we could basically rule out library model sharing because they would clash. That might be safer than saying it _probably_ won't be a hash collision. I think I can add debug checking for models with a view type collision being used in the same adapter at runtime though, and alert the user to use a manual id in that case, so that takes the risk away from random unknown collisions.

What do you think?